### PR TITLE
Upgrades Certifi and removes unused dependencies

### PR DESCRIPTION
**Pull request by RamosFe**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/24568|

## Description
- Upgrade the certifi package to the version 2024.07.04 or higher.
 - Remove zipp and importlib-metadata dependencies.

### CODE REVIEW

**Review by GGP1 — APPROVED** [review]
LGTM

> Authd integration tests failed with the error `RuntimeError: The API was not started as expected.` but the changes introduced aren't related to it, the tests were executed successfully [here](https://github.com/wazuh/wazuh/issues/24568#issuecomment-2304723112).

**Review by fdalmaup — CHANGES_REQUESTED** [review]

File: src/Makefile [hunk 1]
```diff
@@ -1341,7 +1341,7 @@ TAR := tar -xf
 GUNZIP := gunzip
 GZIP := gzip
 CURL := curl -so
-DEPS_VERSION = 30
+DEPS_VERSION = 30-24568
```
[the remark this answers is not in the document]
Comment (RamosFe) [reply src/Makefile]: Fixed in b15c553175aced7f9b37c1460462e32e4030426f.

File: framework/requirements.txt [hunk 2]
```diff
@@ -95,5 +94,4 @@ websocket-client==0.57.0
 Werkzeug==3.0.3
 xmltodict==0.12.0
 yarl==1.7.0
-zipp==3.3.2
 content_size_limit_asgi==0.1.5
```
[the remark this answers is not in the document]
Comment (RamosFe) [reply framework/requirements.txt]: Fixed in b15c553175aced7f9b37c1460462e32e4030426f.

**Review by fdalmaup — APPROVED** [review]
LGTM!

> [!NOTE]

**Pull request merged**